src/app/hooks/useGlobalChat.ts

```diff
@@ -0,0 +1,12 @@
+'use client';
+import { useContext } from 'react';
+import { ChatContextValue } from '../providers/ChatProvider/provider';
+import { ChatContext } from '../providers/ChatProvider/provider';
+
+export const useGlobalChat = (): ChatContextValue => {
+  const context = useContext(ChatContext);
+  if (!context) {
+    throw new Error('useGlobalChat must be used within a ChatProvider');
+  }
+  return context;
+};
```

Comment:
should be called useChat. If its within a context that wraps the whole app of course its global..